packages/jsii/bin/jsii.ts

```diff
@@ -57,12 +57,15 @@ const warningTypes = Object.keys(enabledWarnings);
 
   const compiler = new Compiler({
     projectInfo,
-    watch: argv.watch,
     projectReferences: argv['project-references'],
     failOnWarnings: argv['fail-on-warnings'],
   });
 
-  return { projectRoot, emitResult: await compiler.emit() };
+  const result = argv.watch
+    // When watching, return a promise that never resolves... #magic
+    ? await compiler.watch().then(watch => watch.block())
```

Comment:
I'd still rather this looked like `await compiler.watch()` and the non-blocking version used by tests is `await compiler.watch({ nonblock: true });`

Reply:
Did something of the likes... I get a few more options :)